pkg/helm/ownerreferences.go

```diff
@@ -1,46 +1,100 @@
 package helm
 
 import (
+	"bytes"
+	"io"
 	"strings"
 
-	"k8s.io/apimachinery/pkg/api/meta"
+	"gopkg.in/yaml.v3"
+	"helm.sh/helm/v3/pkg/postrender"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
-	"k8s.io/cli-runtime/pkg/resource"
 )
 
 const (
 	AnnotationPrimaryResource     = "operator-sdk/primary-resource"
 	AnnotationPrimaryResourceType = "operator-sdk/primary-resource-type"
 )
 
-// addOwnerReferenceVisitor returns a visitor function that adds the specified
-// OwnerReference to each resource it visits
-func addOwnerReferenceVisitor(ownerReference metav1.OwnerReference, istioNamespace string) resource.VisitorFunc {
-	return func(info *resource.Info, err error) error {
+// NewOwnerReferencePostRenderer creates a Helm PostRenderer that adds the
+// specified OwnerReference to each rendered manifest
+func NewOwnerReferencePostRenderer(ownerReference metav1.OwnerReference, ownerNamespace string) postrender.PostRenderer {
+	return OwnerReferencePostRenderer{
+		ownerReference: ownerReference,
+		ownerNamespace: ownerNamespace,
+	}
+}
+
+type OwnerReferencePostRenderer struct {
+	ownerReference metav1.OwnerReference
+	ownerNamespace string
+}
+
```

Comment:
nit: maybe add something like `var _ postrender.PostRenderer = &OwnerReferencePostRenderer{}` to guarantee we implement the interface?

Reply:
Not really needed, because `NewOwnerReferencePostRenderer` returns the interface - if `OwnerReferencePostRenderer` doesn't implement this interface, we get a compile-time error.

But maybe it makes sense to make it more explicit by adding a line like you suggest. Thoughts?

Reply:
Yeah, I agree it's easier to see the intention. Also, someone might change the return type of `NewOwnerReferencePostRenderer`, which would cause us to lose the interface implementation check.

Added.